inc/hclib-async.h

```diff
@@ -64,7 +64,9 @@ inline void call_lambda(T* lambda) {
 	const int wid = current_ws()->id;
 	MARK_BUSY(wid);
 	(*lambda)();
-	HC_FREE((void*) lambda);
+    // FIXME - this whole call chain is kind of a mess
```

Comment:
There are a couple of these comments, but I'm not clear on what qualifies as a mess? Do you mean it needs more documentation, or is there something specifically objectionable about this region of code?